practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useState } from 'react';
+import './FormValidation.css';
+import { InputValidate, Button } from '@components';
+import { BUTTON_VARIANTS } from '@constants';
+import { Validation } from '@helpers';
+
+const FormValidation = (props) => {
+  const { handleShowForm } = props;
+  const [formData, setFormData] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+    gender: 'male',
+  });
+
+  const [errors, setErrors] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+  });
+
+  const handleChange = useCallback((e) => {
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [e.target.name]: e.target.value,
+    }));
+  }, []);
```

Comment:
missing depence

Reply:
I added dependencies setFormData because if value of setFormData change, the function handleChange will activate
![image](https://github.com/datdragon777/React-Training/assets/74806826/39b402b6-1aa2-4102-9584-bf42186b93f2)